dvc/objects/__init__.py

```diff
@@ -7,10 +8,25 @@
 logger = logging.getLogger(__name__)
 
 
-def save(odb, obj, **kwargs):
+def save(odb, obj, jobs=None, **kwargs):
     if isinstance(obj, Tree):
-        for _, entry in Tqdm(obj):
-            odb.add(entry.path_info, entry.fs, entry.hash_info, **kwargs)
+        with ThreadPoolExecutor(max_workers=jobs) as executor:
+            for future in Tqdm(
+                as_completed(
+                    executor.submit(
+                        odb.add,
+                        entry.path_info,
+                        entry.fs,
+                        entry.hash_info,
+                        **kwargs
+                    )
+                    for _, entry in obj
```

Comment:
Seems like you could `executor.map`, or am I missing something?

Reply:
We need to manually unpack some attributes of entry (`entry.path_info`, `entry.fs` etc) + add `**kwargs` to the calls. In theory we can do something like `executor.map(lambda entry: odb.add(...), ...)`, what would be the advantage with map though? It would just block the progress bar if there is some big file in the middle, which might not be the best UI.

Reply:
Seemed nicer and shorter. E.g. see how we wrap the worker in tqdm in dvc/objects/stage.py, where we use it in map, so it doesn't block the pbar.